### ISSUE TITLE
ENH: Is possible use rename for flatten MultiIndex?

### ISSUE THREAD

**Issue by jesrael** [pandas-dev/pandas#34254]
From [SO](https://stackoverflow.com/a/61891863/2901002):

    df1 = (df.set_index(['Animal', df.groupby('Animal').cumcount().add(1)])
             .unstack()
             .sort_index(axis=1, level=1))
    df1.columns = [f'{a}_{b}' for a, b in df1.columns]
    df1 = df1.reset_index()
    print (df1)
      Animal  Age_1 Color_1 Length_1  Age_2 Color_2 Length_2  Age_3 Color_3  \
    0    Cat      1   Brown     50cm      2   White     60cm      3   Brown   
    1    Dog      1   White     99cm      2   White    129cm      3   White   
    
      Length_3  
    0     55cm  
    1    105cm  

If I want chain flatten `MultIindex` for one line solution, is it possible? Now `f-string`s cannot chain with `sort_index` and `reset_index` (or not idea how).

Is possible implemented [`MultiIndex.to_flat_index`](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.MultiIndex.to_flat_index.html) for `DataFrame` ?

    df1 = (df.set_index(['Animal', df.groupby('Animal').cumcount().add(1)])
             .unstack()
             .sort_index(axis=1, level=1)
             .to_flat_index(axis=1)
             .reset_index())

**Comment by TomAugspurger** [comment]
Can you add the code for defining `df1`?

The request here is for an `NDFrame.to_flat_index`, with the behavior of

```
df.index = df.index.to_flat_index()  # or .columns for axis=1
```

correct?

**Comment by jesrael** [comment]
Yes, something like [`MultiIndex.droplevel`](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.MultiIndex.droplevel.html) is alos implemented [`Series.droplevel`](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.droplevel.html) and [`DataFrame.droplevel`](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.droplevel.html)
